components/raftstore/src/store/worker/region.rs

```diff
@@ -389,18 +444,7 @@ where
         let start_key = keys::enc_start_key(&region);
         let end_key = keys::enc_end_key(&region);
         check_abort(&abort)?;
-
-        // clear up origin data.
-        let overlap_ranges = self
-            .pending_delete_ranges
-            .drain_overlap_ranges(&start_key, &end_key);
-        if !overlap_ranges.is_empty() {
-            CLEAN_COUNTER_VEC
-                .with_label_values(&["overlap-with-apply"])
-                .inc();
-            self.cleanup_overlap_regions(overlap_ranges)?;
-        }
-        self.delete_all_in_range(&[Range::new(&start_key, &end_key)])?;
+        self.clean_overlap_ranges(start_key, end_key)?;
```

Comment:
blobs of Titan are not cleaned here

Reply:
I think it's difficult to do that. We want to avoid deleting blobs when the rocksdb key still exists. But the ranges that are safe to perform delete_files on have already been removed when we finished `self.delete_all_in_range`.